gravitee-apim-console-webui/src/portal/customization/api/api-list/portal-api-list.component.ts

```diff
@@ -0,0 +1,317 @@
+        switchMap((data: ApiPortalHeaderEditDialogResult) => this.environmentApiHeadersService.createApiHeader(data)),
+        takeUntil(this.unsubscribe$),
+      )
+      .subscribe();
+  }
+
+  public updateHeader(header: ApiPortalHeader): void {
+    this.matDialog
+      .open<ApiPortalHeaderEditDialogComponent, ApiPortalHeaderDialogData, ApiPortalHeaderEditDialogResult>(
+        ApiPortalHeaderEditDialogComponent,
+        {
+          data: { name: header.name, value: header.value },
+          width: GIO_DIALOG_WIDTH.SMALL,
+        },
+      )
+      .afterClosed()
+      .pipe(
+        filter((data): boolean => !!data),
+        switchMap((headerDialogResult: ApiPortalHeaderEditDialogResult) => {
+          const payload: ApiPortalHeader = {
+            ...header,
+            name: headerDialogResult.name,
+            value: headerDialogResult.value,
+          };
+          return this.environmentApiHeadersService.updateApiHeader(payload);
+        }),
+        tap(() => {
+          this.snackBarService.success('API Information updated successfully');
+        }),
+        catchError(({ error }) => {
+          this.snackBarService.error(error.message);
+          return EMPTY;
+        }),
+        takeUntil(this.unsubscribe$),
+      )
+      .subscribe();
+  }
+
+  public deleteHeader(header: ApiPortalHeader): void {
+    this.matDialog
+      .open<GioConfirmDialogComponent, GioConfirmDialogData, boolean>(GioConfirmDialogComponent, {
+        data: {
+          title: 'Delete API Information',
+          content: 'Are you sure you want to delete this API Information?',
+          confirmButton: 'Delete',
+        },
+        role: 'alertdialog',
+        id: 'deletePortalHeaderConfirmDialog',
+      })
+      .afterClosed()
+      .pipe(
+        filter((confirm: boolean): boolean => confirm),
+        switchMap(() => this.environmentApiHeadersService.deleteApiHeader(header)),
+        tap(() => {
+          this.snackBarService.success(`API Information ${header.name} deleted successfully`);
+        }),
+        catchError(({ error }) => {
+          this.snackBarService.error(error);
+          return EMPTY;
+        }),
+        takeUntil(this.unsubscribe$),
+      )
+      .subscribe();
+  }
+
+  private initializeForm() {
+    this.form = new FormGroup<ApiForm>({
+      apiKeyHeader: new FormControl<string>(this.settings.portal.apikeyHeader),
+    });
+    this.formInitialValues = this.form.getRawValue();
+
+    const isReadOnly = !this.permissionService.hasAnyMatching(['environment-api-u']);
+    if (isReadOnly) {
+      this.form.disable();
+    }
+  }
+
+  reset() {
+    this.form.reset();
+    this.initializeForm();
+  }
+
+  submit() {
+    const updatedSettingsPayload: PortalSettings = {
+      ...this.settings,
+      portal: {
+        ...this.settings.portal,
+        apikeyHeader: this.form.controls.apiKeyHeader.value,
+      },
+    };
+
+    this.portalSettingsService
```

Comment:
Probably should do a get then update for the settings

Reply:
I saw example with get before update:
https://github.com/gravitee-io/gravitee-api-management/blob/master/gravitee-apim-console-webui/src/management/settings/api-quality-rules/api-quality-rules.component.ts#L178

<img width="1211" alt="Screenshot 2024-09-09 at 19 49 44" src="https://github.com/user-attachments/assets/800c9371-e5f3-471a-95db-75ac5dc8c04e">

Reply:
Done